float64[3] VecForward 